File: article-merger.py
import os

folder_path = r'C:\Users\Svene\.dev\learning\c-lang'
output_file = 'README.md'

def merge_md_files(folder_path, output_file):
    # Get a list of all .md files in the folder
    md_files = [file for file in os.listdir(folder_path) if file.endswith('.md')]

    # Sort the files to ensure a specific order, if needed
    md_files.sort()

    # Open the output file in write mode
    with open(output_file, 'w', encoding='utf-8') as output:
        # Iterate through each .md file and write its content to the output file
        for md_file in md_files:
            md_file_path = os.path.join(folder_path, md_file)
            with open(md_file_path, 'r', encoding='utf-8') as input_file:
                output.write(f'# {md_file}\n\n')  # Add a header for each file
                output.write(input_file.read())
                output.write('\n\n')  # Add extra newline between files

    print(f'Merged {len(md_files)} .md files into {output_file}')

# Call the function with the specified folder path and output file name
merge_md_files(folder_path, output_file)

File: article-merger.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>
#include <windows.h>

void mergeMarkdownFiles(const char *folderPath, const char *outputFileName)
{
    FILE *outputFile = fopen(outputFileName, "w");

    if (outputFile == NULL)
    {
        perror("Error opening output file");
        exit(EXIT_FAILURE);
    }

    char searchPath[MAX_PATH];
    snprintf(searchPath, sizeof(searchPath), "%s\\*.md", folderPath);

    WIN32_FIND_DATA findFileData;
    HANDLE hFind = FindFirstFile(searchPath, &findFileData);

    if (hFind == INVALID_HANDLE_VALUE)
    {
        perror("Error listing files in the folder");
        exit(EXIT_FAILURE);
    }

    do
    {
        char currentFilePath[MAX_PATH];
        snprintf(currentFilePath, sizeof(currentFilePath), "%s\\%s", folderPath, findFileData.cFileName);

        FILE *currentFile = fopen(currentFilePath, "r");

        if (currentFile == NULL)
        {
            perror("Error opening a file for reading");
            exit(EXIT_FAILURE);
        }

        char fileLine[256];
        while (fgets(fileLine, sizeof(fileLine), currentFile) != NULL)
        {
            fputs(fileLine, outputFile);
        }

        fclose(currentFile);

    } while (FindNextFile(hFind, &findFileData) != 0);

    FindClose(hFind);
    fclose(outputFile);
}

int main()
{
    const char *folderPath = "C:\\Users\\Svene\\.dev\\learning\\c-lang"; // Replace with your actual folder path
    const char *outputFileName = "all-overview.md";

    mergeMarkdownFiles(folderPath, outputFileName);

    printf("Merged Markdown files into %s\n", outputFileName);

    return 0;
}
